### PR TITLE
feat: add args core schema

### LINKED ISSUE #107: JSON Schema for commands of console app
There are a lot of commands in console app of different types, distributed in different ways. But all of them is similar of content. 

For example: 
- all arguments have a prefix `-` or `--`
- most everybody has `help` and `version` commands
- e.t.c

This enhancement will unify how command line arguments are described.

JSON Schema must have following properties: 

| Property    | Type     | Required | Description |
| -           | :-:      | :-:      | -           |
| id          | string   | -        | app id, case insensitive unique identifier, required only if the app name is not unique |
| name        | string   | +        | Case sensitive official engine name, may not be unique |
| delimiters  | string   | +        | String containing delimiters used between command-line options and their arguments, the most commonly used delimiters are " "(space) and "=" |
| arguments | object | + | Schema of command line arguments | 

Each argument has: 

| Property    | Type     | Required | Description |
| -           | :-:      | :-:      | -           |
| id          | string   | -        | The name of the argument from the original tool |
| type        | string   | +        | The value type of the argument. Example: `null`, `number`, `string`  |
| description | string   | +        | Help text from the original tool for that argument, two sentences maximum |
| default     | any      | -        | The default value for that argument; if the argument is read-only and a default is not set then `0`, `""` or `null` is used |
| enum        | array    | -        | Need to use if the argument value has a limited list. Type of argument must be string   |

Also, this schema needed in our repo `usage-parser`

### PR DESCRIPTION
**Pull request by romanrostislavovich**
added `args.json` core schema

Closes #107

### CODE REVIEW

File: src/schema/args.json [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                        },
+                        {
+                            "$ref": "#/definitions/id:linterhub"
+                        },
+                        {
+                            "$ref": "#/definitions/id:arg"
+                        }
+                    ]
+                },
+                "hideName": {
```
Comment (xferra) [inline src/schema/args.json]: `named` seems to be a better wording, what do u think?

**Review by xferra — DISMISSED** [review]
What is the reason to repeat official schema definitions there?

**Review by xferra — CHANGES_REQUESTED** [review]
What is the reason to repeat official schema definitions there?

**Comment by romanrostislavovich** [comment]
If I remove it, we can't to validate arguments later

File: src/schema/args.json (hunk 1 above)
Comment (romanrostislavovich) [reply src/schema/args.json]: I rename it, thx

**Comment by xferra** [comment]
Well, then why they were not listed in the previous version? (in rm repo)

**Comment by romanrostislavovich** [comment]
It's [here](https://github.com/repometric/linterhub/blob/master/schema/args.json#L25)

**Comment by xferra** [comment]
I mean why do we need to repeat [types, enum definitions, etc.](https://github.com/linterhub/schema/pull/144/files#diff-bd55d66a1d79b8cc07468c0f7f485f89R61) that are already specified in JSON Schema?

**Comment by romanrostislavovich** [comment]
I can remove it

File: src/schema/args.json [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                        },
+                        {
+                            "$ref": "#/definitions/id:linterhub"
+                        },
+                        {
+                            "$ref": "#/definitions/id:arg"
+                        }
+                    ]
+                },
+                "Named": {
```
Comment (xferra) [inline src/schema/args.json]: Uppercase?

File: src/schema/args.json [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                    "title": "Named",
+                    "description": "Hide orignal name. Used when package have multiple arguments without prefix.",
```
Comment (xferra) [inline src/schema/args.json]: Please be careful, we never use dot at the end

File: src/schema/args.json [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            "type": "string",
+            "pattern": "^linterhub:[a-z]+$"
+        },
+        "id:arg": {
+            "title": "Id arg",
+            "description": "Id with prefix arg",
+            "type": "string",
+            "pattern": "^arg:[-a-z]+$"
+        },
+        "argument": {
```
Comment (xferra) [inline src/schema/args.json:35]: Then argument it's just a property as defined in json schema, isn't it?

Comment (romanrostislavovich) [reply src/schema/args.json:35]: Yes

File: src/schema/args.json (hunk 3 above)
Comment (romanrostislavovich) [reply src/schema/args.json]: I will fix it

File: src/schema/args.json (hunk 2 above)
Comment (romanrostislavovich) [reply src/schema/args.json]: I will fix it

File: src/schema/args.json (hunk 4 above)
Comment (xferra) [reply src/schema/args.json:35]: Still, why do we need to duplicate it's definition if we can reuse it from JSON Schema?

Comment (xferra) [reply src/schema/args.json:35]: Ah, we have custom definition for`id` and extra properties.

Comment (romanrostislavovich) [reply src/schema/args.json:35]: I'm not sure, but we don't have duplicate it. Maybe you mean a create separate JSON Schema for it?

Comment (xferra) [reply src/schema/args.json:35]: I mean that this is a schema that defines a new args schema. Could you please test whether it's actually is able to validate args json? If it's still is not clear - ping me directly.

**Comment by xferra** [comment]
@svoboda-rabstvo Please check my variant #165

**Comment by romanrostislavovich** [comment]
I will close it. Finished in #165

**Pull request closed without merging**